katas/langtonant/charles/langtonant.js

```diff
@@ -0,0 +1,82 @@
+const arg1 = process.argv[2];//N dimension grid
+const arg2 = process.argv[3];//N steps
+
+let gridArray = new Array(arg1);
+let rowPositionStep;
+let columnPositionStep;
+let pointer;
+let isPainted = false;
+let startPosition;
+
+const directionRight = '[r]';
+const directionLeft = '[l]';
+const directionUp = '[u]';
+const directionDown = '[d]';
+
+const startStepDirection = directionUp;
+
+function createGrid() {
+    startPosition = arg1 % 2 == 0 ? (arg1) / 2 - 1 : (arg1) / 2 - 0.5;
+    for (let indexRow = 0; indexRow < arg1; indexRow++) {
+        gridArray[indexRow] = new Array(arg1);
+        for (let indexColumn = 0; indexColumn < arg1; indexColumn++) {
+            gridArray[indexRow][indexColumn] = (indexRow == startPosition && indexColumn == startPosition) ? startStepDirection : '   ';
+        }
+    }
+}
+
+function directionNextStep(rowPosition, columnPosition) {
+    switch (gridArray[rowPosition][columnPosition]) {
+    case directionUp:
+        pointer = rowPosition - 1;
```

Comment:
For each case, we have some repeated code that could be improved by adding a function that parameterizes the values that change on every situation, consider that for your future implementations